utils/s3/client.cc

```diff
@@ -84,34 +88,57 @@ future<> ignore_reply(const http::reply& rep, input_stream<char>&& in_) {
 client::client(std::string host, endpoint_config_ptr cfg, semaphore& mem, global_factory gf, private_tag, std::unique_ptr<aws::retry_strategy> rs)
         : _host(std::move(host))
         , _cfg(std::move(cfg))
+        , _creds_invalidation_timer([this] {
+            s3l.info("Credentials update attempt in background failed. Outdated credentials will be discarded, triggering synchronous re-obtainment"
+                     " attempts for future requests.");
+            _credentials = {};
+        })
+        , _creds_update_timer([this] {
+            std::ignore = [this]() -> future<> {
+                s3l.info("Update creds in the background");
+                _credentials = co_await _creds_provider_chain.get_aws_credentials();
+                _creds_invalidation_timer.rearm(_credentials.expires_at);
+                _creds_update_timer.arm(_credentials.expires_at - 1h);
+            }();
+        })
+        , _creds_sem(1)
         , _gf(std::move(gf))
         , _memory(mem)
-        , _retry_strategy(std::move(rs))
-{
+        , _retry_strategy(std::move(rs)) {
+    _creds_provider_chain
+        .add_credentials_provider(std::make_unique<aws::config_file_aws_credentials_provider>(db::config::get_conf_sub("object_storage.yaml").native()))
+        .add_credentials_provider(std::make_unique<aws::environment_aws_credentials_provider>());
+
+    _creds_update_timer.arm(seastar::lowres_clock::now());
 }
 
 void client::update_config(endpoint_config_ptr cfg) {
     if (_cfg->port != cfg->port || _cfg->use_https != cfg->use_https) {
         throw std::runtime_error("Updating port and/or https usage is not possible");
     }
     _cfg = std::move(cfg);
+    _credentials = {};
```

Comment:
Also need to
* disarm creds-invalidation-timer
* re-arm creds-update-timer for now()